docs/guide/writing-tasks.md

```diff
@@ -250,10 +250,10 @@ the same organization cannot re-use them. `qwerty239abc` from the example above
 variable, it's just an internal ID. No one can brute force your secrets from such ID. In addition, Cirrus CI doesn't know
 a relation between an encrypted variable and a repository for which the encrypted variable was created.
 
-## Matrix Modification
+## Matrix Modifications
 
 Sometimes it's useful to run the same task against different software versions. Or run different batches of tests based
-on an environment variable. For cases like these `matrix` modification comes very handy. It's possible to use `matrix`
+on an environment variable. For cases like these, the `matrix` modifier comes very handy. It's possible to use `matrix`
```

Comment:
Do you think we can keep the title as *Matrix Modification* since it's an intention of an action and use *modifier* here. I'm not a native speaker so it's hard for me to judge if it sounds right. 😅 

Internally `matrix` is just used while parsing to perform an action of duplicating a task YAML node it's in into several task nodes.

Reply:
Yeah matrix modification sounds the best IMO.  
My native language is Swedish, but I've grown up speaking it and English.

Reply:
Well...I'm a native speaker and I can't do any language, even English :smile: 

`Modification` is the action (or state).   

> I made the modification

or 

> The modification is complete

Whereas `Modifier` is the "thing", the literal `matrix:` in the yaml.

Reply:
To make it more confusing, there's also 'modified' which is exactly like 'modification' except it already happened, unless you're talking about the future.   Dang who the heck invented this crap language...they should have left it up to programmers :smile:

Reply:
haha, `Grammar grammar = new Grammar(isGood = False);`

Reply:
> the `matrix` modifier

should defiantly stay. But what about keeping title as *Matrix Modification* (singular) to keep the links untouched?

Reply:
Agreed